process_report/process_report.py

```diff
@@ -1,9 +1,67 @@
 import argparse
 import os
+import sys
 
+import json
 import pandas
 
 
+### Invoice field names
+INVOICE_DATE_FIELD = 'Invoice Month'
+PROJECT_FIELD = 'Project - Allocation'
+PROJECT_ID_FIELD = 'Project - Allocation ID'
+PI_FIELD = 'Manager (PI)'
+INVOICE_EMAIL_FIELD = 'Invoice Email'
+INVOICE_ADDRESS_FIELD = 'Invoice Address'
+INSTITUTION_FIELD = 'Institution'
+INSTITUTION_ID_FIELD = 'Institution - Specific Code'
+SU_HOURS_FIELD = 'SU Hours (GBhr or SUhr)'
+SU_TYPE_FIELD = 'SU Type'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def get_institution_from_pi(institute_map, pi_uname):
+    institution_key = pi_uname.split('@')[-1]
+    institution_name = institute_map.get(institution_key, '')
+
+    if institution_name == '':
+        print(f"Warning: PI name {pi_uname} does not match any institution!")
+    
+    return institution_name
+
+
+def load_institute_map() -> dict:
+    with open('institute_map.json', 'r') as f:
+        institute_map = json.load(f)
+
+    return institute_map
+
+
+def load_old_pis(old_pi_file):
+    old_pi_dict = dict()
+
+    try:
+        with open(old_pi_file) as f:
+            for pi_info in f: 
+                pi, first_month = pi_info.strip().split(',')
+                old_pi_dict[pi] = first_month
+    except FileNotFoundError:
+        print('Applying credit 0002 failed. Old PI file does not exist')
+        sys.exit(1)
+    
+    return old_pi_dict
+
+
+def is_old_pi(old_pi_dict, pi, invoice_month):
+    if pi in old_pi_dict and old_pi_dict[pi] != invoice_month: 
```

Comment:
This will not work if you are processing old invoices. Say you want to process invoices for `2024-02` that is 2 months ago. A PI with first invoice of `2024-03` is going to appear as an old PI.

I'm okay with fixing this in a follow-up PR while introducing a mechanism to update the PI file.